spec/test-outputs/www-integration.out.vcl

```diff
@@ -382,6 +386,12 @@ sub vcl_fetch {
     return (pass);
   }
 
+  # Pass through good responses to POST requests (i.e. cache responses to POST
+  # requests that are the result of client error).
+  if (req.request == "POST" && !(beresp.status >= 400 && beresp.status <= 499)) {
+    return (pass);
```

Comment:
Maybe I'm missing something - but I'm not sure what this actually does as a change. As I understand it tells varnish not to cache POST requests that aren't in 4xx status. But not caching is the default behaviour I'm not sure what actually tells this to cache them.

Reply:
As far as I'm aware, the behaviour without an explicit `return` is to fall through to the overridden function, and the [default behaviour in that](https://github.com/varnish/Varnish-Cache/blob/2.1/bin/varnishd/default.vcl#L114) is `return (deliver)`. But I'm not exactly a VCL / Varnish expert, so my understanding my well be flawed...

Reply:
> But I'm not exactly a VCL / Varnish expert, so my understanding my well be flawed...

That's my position too, so take my comments with a pinch of salt 😄 

Looking again I see that this change you have https://github.com/alphagov/govuk-cdn-config/pull/106/files#diff-c789eaeb1ab164ae6d3fb552b5b855d7L214 is the bit that I was missing. That previously prevented the cache lookup.

From my frantic googling to understand this a common theme seems to be that by passing a POST to lookup we end up with [varnish sending a GET request to the backend regardless of which method the requester](https://docs.varnish-software.com/tutorials/caching-post-requests/#step-4-make-sure-the-backend-gets-a-post-request) used. I'm not sure if this behaviour is still current but thought worth mentioning.

Reply:
That's an interesting guide that I'd not seen before. I need to think about this some more...